internal/servers/worker/proxy/proxy.go

```diff
@@ -15,6 +15,11 @@ import (
 // a provided ClientConn and the RemoteEndpoint, as well as the parameters to update
 // the connection in the connection repository.
 type Config struct {
+	// UserClientIp is the user's client IP
+	UserClientIp string
```

Comment:
Should this be a `net.IP` at this point, given client address is `net.TCPAddr`?

Reply:
I'm not sure []byte is significanlty better than string.  ClientAddress makes sense a net.TCPAddr since there's a port with the IP.

Reply:
Using a `net.IP` guarantees that it is a valid address (if created via correct means). We can always convert later if we want to do checks, but by that argument, shouldn't we use a string for ClientAddress since it's just `ip:port`.

To me it just feels weird to have a typed value and a parsed value next to each other when they represent very similar things at the same point in the code path.

Reply:
Add the requested changes in 241d222d1
Looking at the changes I'm not sure it adds any value, but happy to resolve this.

Reply:
BTW, I can admit that it looks better.